src/libraries/System.Console/src/System/IO/ConsoleStream.cs

```diff
@@ -90,5 +91,37 @@ protected void ValidateWrite(byte[] buffer, int offset, int count)
                 throw Error.GetWriteNotSupported();
             }
         }
+
+        protected static void AccumulateNewLines(StringBuilder accumulator, ReadOnlySpan<char> buffer, Action<string> printer)
```

Comment:
if this method is specifc to iOS and not reused anywhere else, why not to keep it in `ConsolePal.iOS.cs` and replace `Action<string> printer` with direct calls to `Interop.Sys.Log`?

Reply:
Yeah I think we should move it to ConsolePal.iOS.cs since we no longer need it on Android.

We should also implement Rolf's suggestion from https://github.com/dotnet/runtime/pull/37078#discussion_r431814922 to walk the buffer backwards.